### PR TITLE
Devs want to show/hide attachment button depends on site configuration

### PR DESCRIPTION
**Pull request by DavDo**
**Jira issue:**
https://glia.atlassian.net/browse/MOB-1191?atlOrigin=eyJpIjoiOTZlOTE1YjFmMjAxNDhiOThiMDY0NmZhZTI2MDc2OGEiLCJwIjoiaiJ9

MOB-1191

### CODE REVIEW

**Review by gugalo — APPROVED** [review]

**Review by AndriiHorishniiMOC — APPROVED** [review]

**Review by deniss-mc — CHANGES_REQUESTED** [review]
Change target branch to develop (rebase)

**Comment by deniss-mc** [comment]
This will need to be combined with the Configuration approach at some point @jakob-mc 

At the moment Looks OK

@dukhovnyi  what should happen if information query will fail? does it need to retry? should it default to some parameters?

**Review by deniss-mc — APPROVED** [review]

**Comment by dukhovnyi** [comment]
> This will need to be combined with the Configuration approach at some point CC: @jakob-mc
> 
> At the moment Looks OK
> 
> @dukhovnyi what should happen if information query will fail? does it need to retry? should it default to some parameters?
> 
> We would like to move configuration initialization out of Activity classes at some point - this way we will know if we can put callback to application level or still need to query those each time we enter the Glia Activities

@deniss-mc , in case of failed request we need to show the button and give the possibility to visitors to try to upload files. In case it is unavailable, visitors get the error message as it works now. Does it make sense?
cc: @DavDo

File: widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java [hunk 1]
```diff
@@ -251,7 +256,7 @@ public ChatController(
                 .setChatItems(new ArrayList<>())
                 .setLastTypedText("")
                 .setChatInputMode(ChatInputMode.ENABLED_NO_ENGAGEMENT)
-                .setIsAttachmentButtonVisible(false)
+                .setIsAttachmentButtonNeeded(false)
```
Comment (deniss-mc) [inline widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java:259]: id still call it visible :)

**Comment by deniss-mc** [comment]
> > This will need to be combined with the Configuration approach at some point CC: @jakob-mc
> > At the moment Looks OK
> > @dukhovnyi what should happen if information query will fail? does it need to retry? should it default to some parameters?
> > We would like to move configuration initialization out of Activity classes at some point - this way we will know if we can put callback to application level or still need to query those each time we enter the Glia Activities
> 
> @deniss-mc , in case of failed request we need to show the button and give the possibility to visitors to try to upload files. In case it is unavailable, visitors get the error message as it works now. Does it make sense? cc: @DavDo

I ment if the SiteInfo request from hub fails (there could be more configuration options like white label thingy, call disable etc. endless possibilities :) ). In case of attachments it should default to 'show' then - at the moment I see

`siteInfo == null` = don't show

**Comment by DavDo** [comment]
> > This will need to be combined with the Configuration approach at some point CC: @jakob-mc
> > At the moment Looks OK
> > @dukhovnyi what should happen if information query will fail? does it need to retry? should it default to some parameters?
> > We would like to move configuration initialization out of Activity classes at some point - this way we will know if we can put callback to application level or still need to query those each time we enter the Glia Activities
> 
> @deniss-mc , in case of failed request we need to show the button and give the possibility to visitors to try to upload files. In case it is unavailable, visitors get the error message as it works now. Does it make sense? cc: @DavDo

@dukhovnyi In case of request with **App Token** we always receive null for SiteInfo that for I think we need to hide button only when `isVisitorAllowed() == false`

**Review by deniss-mc — APPROVED** [review]

**Pull request merged**